src/Symfony/Component/ErrorHandler/ErrorException.php

```diff
@@ -0,0 +1,472 @@
+        $this->charset = $charset ?: ini_get('default_charset') ?: 'UTF-8';
+        $this->fileLinkFormat = $fileLinkFormat;
+    }
+
+    /**
+     * Sets the format for links to source files.
+     *
+     * @param string|FileLinkFormatter $fileLinkFormat The format for links to source files
+     *
+     * @return string The previous file link format
+     */
+    public function setFileLinkFormat($fileLinkFormat)
+    {
+        $old = $this->fileLinkFormat;
+        $this->fileLinkFormat = $fileLinkFormat;
+
+        return $old;
+    }
+
+    /**
+     * Sends the error associated with the given Exception as a plain PHP response.
+     *
+     * This method uses plain PHP functions like header() and echo to output
+     * the response.
+     *
+     * @param \Exception|FlattenException $exception An \Exception or FlattenException instance
+     */
+    public function sendPhpResponse($exception)
+    {
+        if (!$exception instanceof FlattenException) {
+            $exception = FlattenException::create($exception);
+        }
+
+        if (!headers_sent()) {
+            header(sprintf('HTTP/1.0 %s', $exception->getStatusCode()));
+            foreach ($exception->getHeaders() as $name => $value) {
+                header($name.': '.$value, false);
+            }
+            header('Content-Type: text/html; charset='.$this->charset);
+        }
+
+        echo $this->getHtmlContent($exception);
+    }
+
+    /**
+     * Gets the content associated with the given exception.
+     *
+     * @param \Exception|FlattenException $exception An \Exception or FlattenException instance
+     * @param string                      $format    The request format (html, json, xml, txt)
+     *
+     * @return string The formatted content as a string
+     */
+    public function getContent($exception, string $format = 'html'): string
+    {
+        if (!$exception instanceof FlattenException) {
+            $exception = FlattenException::create($exception);
+        }
+
+        $statusCode = $exception->getStatusCode();
+
+        if (class_exists(Response::class) && isset(Response::$statusTexts[$statusCode])) {
+            $title = Response::$statusTexts[$statusCode];
+        } else {
+            $title = 'Whoops, looks like something went wrong.';
+        }
+
+        switch (strtolower($format)) {
+            case 'json':
+                return $this->getJsonContent($exception, $title, $statusCode);
+            case 'xml':
+            case 'rdf':
+                return $this->getXmlContent($exception, $title, $statusCode);
+            case 'txt':
+                return $this->getTxtContent($exception, $title, $statusCode);
+            default:
+                return $this->getHtmlContent($exception);
+        }
+    }
+
+    /**
+     * Gets the HTML content associated with the given exception.
+     *
+     * @return string The content as a string
+     */
+    public function getHtmlBody(FlattenException $exception)
```

Comment:
isn't it strange that this method (and the one fetching the style) are public but the `getJsonContent`, `getHtmlContent` are private ?